packages/gestalt/src/Status.js

```diff
@@ -0,0 +1,83 @@
+// @flow strict
+import React, { type Node } from 'react';
+import PropTypes from 'prop-types';
+import Box from './Box.js';
+import Icon from './Icon.js';
+import Flex from './Flex.js';
+import Text from './Text.js';
+
+export type StatusColor = 'gray' | 'green' | 'orange' | 'red';
+
+const ICON_COLOR_MAP = {
+  unstarted: {
+    icon: 'workflow-status-unstarted',
+    color: 'darkGray',
+  },
+  inProgress: {
+    icon: 'workflow-status-in-progress',
+    color: 'green',
+  },
+  halted: {
+    icon: 'workflow-status-halted',
+    color: 'darkGray',
+  },
+  ok: {
+    icon: 'workflow-status-ok',
+    color: 'green',
+  },
+  problem: {
+    icon: 'workflow-status-problem',
+    color: 'red',
+  },
+  canceled: {
+    icon: 'workflow-status-canceled',
+    color: 'gray',
+  },
+  warning: {
+    icon: 'workflow-status-warning',
+    color: 'orange',
+  },
+};
+
+type Props = {|
+  type: 'unstarted' | 'inProgress' | 'halted' | 'ok' | 'problem' | 'canceled' | 'warning',
+  title: string,
+  subtext?: string,
+|};
+
+export default function Status({ type, title, subtext }: Props): Node {
+  const { icon, color } = ICON_COLOR_MAP[type];
+
+  return (
+    <Flex direction="column">
```

Comment:
This is beautiful. So simple and clean 😍

Reply:
Yay!